src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -77,7 +77,107 @@ object EdgeSchema {
   def stripEdgeRelation(str: String): String = str.split(sep, 2).last
 }
 
+/**
+ * This class is a wrapper around scala.reflect.api.Type that caches results of calls to the Type so
+ * we don't need to go through reflection for future calls. Caching is done through lazy vals in most
+ * cases but TrieMaps are used in places where results build over time.
+ * @param tpe
+ * @tparam T
+ */
+class CachedType[T](val tpe: Type) {
+  def this()(implicit ttag: TypeTag[T]) = this(typeOf[T])
```

Comment:
we talked about this, but for posterity:
this line declares a no-arg constructor and gets the type from an implicit TypeTag parameter. This class has a type parameter so that the compiler can differentiate between the different CachedType instances. The rest of the code does not care about the type thus the [_] everywhere.